eduHub/src/main/kotlin/com/arsiu/eduhub/controller/nats/assignment/AssignmentNatsControllerDeleteById.kt

```diff
@@ -0,0 +1,67 @@
+package com.arsiu.eduhub.controller.nats.assignment
+
+import com.arsiu.eduhub.controller.nats.NatsController
+import com.arsiu.eduhub.service.AssignmentService
+import com.arsiu.eduhub.v2.assignmentsvc.NatsSubject.ASSIGNMENT_DELETE_BY_ID
+import com.arsiu.eduhub.v2.assignmentsvc.commonmodels.assignment.AssignmentProto.AssignmentResponse
+import com.arsiu.eduhub.v2.assignmentsvc.commonmodels.assignment.AssignmentProto.AssignmentResponse.Failure
+import com.arsiu.eduhub.v2.assignmentsvc.commonmodels.assignment.AssignmentProto.AssignmentResponse.Failure.CustomError
+import com.arsiu.eduhub.v2.assignmentsvc.commonmodels.assignment.AssignmentProto.AssignmentResponse.Success
+import com.arsiu.eduhub.v2.assignmentsvc.input.reqreply.assignment.DeleteByIdAssignmentRequestProto.DeleteByIdAssignmentRequest
+import com.arsiu.eduhub.v2.assignmentsvc.output.reqreply.assignment.DeleteByIdAssignmentResponseProto.DeleteByIdAssignmentResponse
+import com.google.protobuf.Parser
+import io.nats.client.Connection
+import org.springframework.stereotype.Component
+
+@Component
+class AssignmentNatsControllerDeleteById(
+    override val connection: Connection,
+    private val service: AssignmentService
+) : NatsController<DeleteByIdAssignmentRequest, DeleteByIdAssignmentResponse> {
+
+    override val subject: String = ASSIGNMENT_DELETE_BY_ID
+    override val parser: Parser<DeleteByIdAssignmentRequest> = DeleteByIdAssignmentRequest.parser()
+
+    override fun handler(request: DeleteByIdAssignmentRequest): DeleteByIdAssignmentResponse {
+        return try {
```

Comment:
You can try use runCatching instead of try catch
```
return runCatching {
        val id = request.request.assignmentId.id
        getSuccessResponse(service.delete(id))
    }.getOrElse { ex ->
        getFailureResponse(ex.javaClass.simpleName, ex.toString())
    }
```
 This approach is more idiomatic in Kotlin

Reply:
+